### DISCUSSION TITLE
enhancement of create similar table

### DISCUSSION THREAD

**Discussion by tataremka** [pawelsalawa/sqlitestudio#3704] [Ideas]
I have stopped with some problem...
I have many tables (29 exactly) with data that have the same structure.
I can edit structure of the table one at the time, i mean i added some constraints to one of the tables.
I  would like to do the same with the rest of tables... 
The easiest way would be "paste" structure from modified table to the another table...
Is it possible to change structure of any table according to another structure.
Columns with the same name will keep its data but structure would be updated... just as it is now with manual editing... New columns would be just added...
I can of course do that by creating new database, new tables and copy all data, but...
When I am working on database with data in it and want to some small changes, it would be very usefull to have such tool to "paste" structure to the existing data table...
Just 2 clicks and work is done... now i have to spend lots of time to do this...
I have to do that now manually, but in the future, i think it would be nice tool for the designer of database in SQLite Studio... not simple manager :)

**Comment by pawelsalawa** [comment]
What exactly would you like to copy-paste? Constraints? Entire columns?

How would you propose this to be done in the application? What do you select, what copy, where to paste, etc?

**Comment by tataremka** [comment]
... constraints... Paste structure of the table to another table... add constraints to the existing columns and just add columns if thay dont exist...  Now i can create similar table... but i want to do that on the existing table with alraedy existing data...
After editing one table I could just paste modified structure to the rest of tables...
Maybe it is possible in easy steps... but now i can see that it is lots of work with creating tables and copying data...

**Comment by pawelsalawa** [comment]
I don't see how this could be made clear and usable to user. That is, how user would pick which constraints from which columns to copy, then which of copied element apply to which columns...

**Comment by tataremka** [comment]
OK, i am just user that have faced some problem to paste the same structure of table many times...

One more question: can you "pick up constraint from which column to copy, (...)" when you want to "create similar table"?
 That is in the topic... All i wanted to do is just do the same: create similar table, only difference is that i dont want to create new table, i want only make it on the existing table, adding new columns (maybe delete?) or modifying existing without loosing data... So... just "make similar table".
It is ok if it is not possible to do, just i've faced problem i have to solve and it seemed for me doable in that way...

**Comment by pawelsalawa** [comment]
Doing so on an existing table is not simple. There are so many different cases to consider. Tables can be very similar, but they also can be very, very different. It's really hard (not impossible, but close to that) to implement proper behavior in all these cases.
